packages/broker/src/plugins/operator/MaintainTopologyHelper.ts

```diff
@@ -82,68 +67,28 @@ export class MaintainTopologyHelper extends EventEmitter<MaintainTopologyHelperE
-        const sponsorship = new Contract(sponsorshipAddress, sponsorshipABI, this.operatorContract.provider) as unknown as Sponsorship
-        return toStreamID(await sponsorship.streamId())
-    }
-
-    private async pullStakedStreams(requiredBlockNumber: number): Promise<StreamID[]> {
-        const createQuery = (lastId: string, pageSize: number) => {
-            return {
-                query: `
-                    {
-                        operator(id: "${this.operatorContractAddress}") {
-                            stakes(where: {id_gt: "${lastId}"}, first: ${pageSize}) {
-                                sponsorship {
-                                    id
-                                    stream {
-                                        id
-                                    }
-                                }
-                            }
-                        }
-                        _meta {
-                            block {
-                            number
-                            }
-                        }
-                    }
-                    `
-            }
-        }
-        const parseItems = (response: any) => {
-            if (!response.operator) {
-                logger.error('Unable to find operator in The Graph', { operatorContractAddress: this.operatorContractAddress })
-                return []
-            }
-            return response.operator.stakes
-        }
-        this.theGraphClient.updateRequiredBlockNumber(requiredBlockNumber)
-        const queryResult = this.theGraphClient.queryEntities<any>(createQuery, parseItems) // TODO: add type
-
+        this.contractFacade.addOperatorContractStakeEventListener('Unstaked', this.onUnstakedListener)
+        
+        const queryResult = this.contractFacade.pullStakedStreams(latestBlock)
         for await (const stake of queryResult) {
-            const sponsorshipId = stake.sponsorship?.id
-            const streamId = toStreamIDSafe(stake.sponsorship?.stream?.id)
-            // TODO: null-checks needed or being too defensive?
-            if (streamId !== undefined && sponsorshipId !== undefined && this.streamIdOfSponsorship.get(sponsorshipId) !== streamId) {
+            const sponsorshipId = toEthereumAddress(stake.sponsorship.id)
+            const streamId = toStreamID(stake.sponsorship.stream.id)
+            if (this.streamIdOfSponsorship.get(sponsorshipId) !== streamId) {
                 this.streamIdOfSponsorship.set(sponsorshipId, streamId)
                 const sponsorshipCount = (this.sponsorshipCountOfStream.get(streamId) || 0) + 1
                 this.sponsorshipCountOfStream.set(streamId, sponsorshipCount)
             }
         }
-        return Array.from(this.sponsorshipCountOfStream.keys())
+        const initialStreams = Array.from(this.sponsorshipCountOfStream.keys())
+        if (initialStreams.length > 0) {
+            this.emit('addStakedStreams', initialStreams)
+        }
```

Comment:
Or

```ts
if (this.sponsorshipCountOfStream.size) {
    this.emit('addStakedStreams', [...this.sponsorshipCountOfStream.keys()])
}
```
which saves you a redundant keys-to-array conversion for each zero-sized map.

Reply:
Using helper variable makes it bit more readable. Converting an empty set to an empty array should be quite efficient anyway.

Reply:
I suppose you could combine both points like:
```ts
if (this.sponsorshipCountOfStream.size > 0) {
    const initialStreams = Array.from(this.sponsorshipCountOfStream.keys())
    this.emit('addStakedStreams', initialStreams)
}
```

Reply:
Fair point @harbu.

@teogeb, your point about readability is fair, too. I def can't tell what are the keys of `sponsorshipCountOfStream` by the name of the map itself w/o looking it up. And what makes them "initial", eh? For readability why not just

```ts
const queryResult = this.contractFacade.pullStakedStreams(latestBlock)

// this is new
const initialStreamIds: string[] = []

for await (const stake of queryResult) {
    const sponsorshipId = toEthereumAddress(stake.sponsorship.id)
    const streamId = toStreamID(stake.sponsorship.stream.id)
    if (this.streamIdOfSponsorship.get(sponsorshipId) !== streamId) {
        this.streamIdOfSponsorship.set(sponsorshipId, streamId)
        const sponsorshipCount = (this.sponsorshipCountOfStream.get(streamId) || 0) + 1
        this.sponsorshipCountOfStream.set(streamId, sponsorshipCount)

        // and this is new
        initialStreamIds.push(streamId)
    }
}

// readabilitiest ever!
if (initialStreamIds.length) {
    this.emit('addStakedStreams', initialStreamIds)
}
```

Reply:
Applied a simplification